mk/config.mk

```diff
@@ -423,14 +423,22 @@ CFG_DT ?= n
 CFG_DTB_MAX_SIZE ?= 0x10000
 
 # Device Tree Overlay support.
-# This define enables support for an OP-TEE provided DTB overlay.
-# One of two modes is supported in this case:
-# 1. Append OP-TEE nodes to an existing DTB overlay located at CFG_DT_ADDR or
-#    passed in arg2
-# 2. Generate a new DTB overlay at CFG_DT_ADDR
-# A subsequent boot stage must then merge the generated overlay DTB into a main
+# CFG_EXTERNAL_DTB_OVERLAY allows to append a DTB overlay into an existing
+# external DTB. The overlay is created when no valid DTB overlay is found.
+# CFG_GENERATE_DTB_OVERLAY allows to create a DTB overlay at external
+# DTB location.
+# External DTB location (physical address) is provided either by boot
+# argument arg2 or from CFG_DT_ADDR if defined.
+# A subsequent boot stage can then merge the generated overlay DTB into a main
 # DTB using the standard fdt_overlay_apply() method.
 CFG_EXTERNAL_DTB_OVERLAY ?= n
+CFG_GENERATE_DTB_OVERLAY ?= n
+
+ifeq (yy,$(CFG_EXTERNAL_DTB_OVERLAY)$(CFG_GENERATE_DTB_OVERLAY))
+$(error CFG_EXTERNAL_DTB_OVERLAY and CFG_GENERATE_DTB_OVERLAY are exclusive)
+endif
+CFG_USE_DTB_OVERLAY ?= $(call cfg-one-enabled,CFG_EXTERNAL_DTB_OVERLAY \
```

Comment:
It is OK to define a derived variable like this, but it should be prefixed by an underscore (`_CFG_USE_DTB_OVERLAY`). The rationale being, it makes it more clear what is supposed to be set by the user or not; and also they are not exported to `$(out-dir)/conf.mk`.

Reply:
Ok thanks, I wondered if you had such thing. Should I move it to core.mk or let it in config.mk ?

Reply:
You can leave it here. Also better use `:=` rather than `?=` since the reason we're using `?=` everywhere is to consider environment variables (`CFG_FOO=x make` vs. `make CFG_FOO=x` should do the same). That is not applicable here.